applications/GeoMechanicsApplication/tests/cpp_tests/custom_utilities/test_constitutive_law_utilities.cpp

```diff
@@ -72,13 +72,17 @@ KRATOS_TEST_CASE_IN_SUITE(CohesionCanBeFetchedFromUMatParameters, KratosGeoMecha
     KRATOS_EXPECT_DOUBLE_EQ(ConstitutiveLawUtilities::GetCohesion(properties), 2.0);
 
     properties.Erase(INDEX_OF_UMAT_C_PARAMETER);
-    KRATOS_EXPECT_EXCEPTION_IS_THROWN(ConstitutiveLawUtilities::GetCohesion(properties),
-                                      "Material 0 does not have INDEX_OF_UMAT_C_PARAMETER");
+    KRATOS_EXPECT_EXCEPTION_IS_THROWN(
+        ConstitutiveLawUtilities::GetCohesion(properties),
+        "ConstitutiveLawUtilities::GetCohesion failed. There are no GEO_COHESION and underlying "
+        "UMAT Error: There is no INDEX_OF_UMAT_C_PARAMETER for material 0.");
```

Comment:
This is already way clearer than it was before, thanks for picking this up! 

Nitpicking, but I would rephrase the error message slightly (and similar for the friction angle):
```suggestion
        "ConstitutiveLawUtilities::GetCohesion failed. There is no GEO_COHESION available and
        "attempting to get the cohesion from UMAT parameters resulted in the following error: "
        "There is no INDEX_OF_UMAT_C_PARAMETER for material 0.");
```
Curious what you think!

Reply:
@WPK4FEM and myself agree with this suggestion.

Reply:
Richard, thank you for the detailed message. Used it.